### PR TITLE
chore: update e2e tests to Kubernetes v1.27.2

### PR DESCRIPTION
**Pull request by simonpasquier**
## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by simonpasquier** [comment]
Not sure why but I can't find a github action that would spin up a kind cluster with kube 1.27 :shrug:

**Comment by philipgough** [comment]
@simonpasquier looks like 1.27.1 is supported at least https://github.com/kubernetes-sigs/kind/releases

**Comment by simonpasquier** [comment]
I tried it previously with the engineerd/setup-kind action but let's see with helm/kind-action.

**Comment by simonpasquier** [comment]
Same result

```
2023-06-02T06:58:30.3540247Z Creating cluster "control-plane" ...
2023-06-02T06:58:30.3541585Z  • Ensuring node image (kindest/node:v1.27.1) 🖼  ...
2023-06-02T06:58:42.7247221Z  ✓ Ensuring node image (kindest/node:v1.27.1) 🖼
2023-06-02T06:58:42.7248050Z  • Preparing nodes 📦   ...
2023-06-02T06:58:46.4471034Z  ✓ Preparing nodes 📦 
2023-06-02T06:58:46.5135502Z  • Writing configuration 📜  ...
2023-06-02T06:58:46.8322099Z  ✓ Writing configuration 📜
2023-06-02T06:58:46.8322755Z  • Starting control-plane 🕹️  ...
2023-06-02T07:00:45.1087597Z  ✗ Starting control-plane 🕹️
...
2023-06-02T07:00:45.5453767Z ERROR: failed to create cluster: failed to init node with kubeadm: command "docker exec --privileged control-plane-control-plane kubeadm init --skip-phases=preflight --config=/kind/kubeadm.conf --skip-token-print --v=6" failed with error: exit status 1
...
2023-06-02T07:00:45.5661603Z I0602 07:00:44.595368     135 round_trippers.go:553] GET https://control-plane-control-plane:6443/healthz?timeout=10s  in 0 milliseconds
2023-06-02T07:00:45.5662228Z I0602 07:00:45.095230     135 round_trippers.go:553] GET https://control-plane-control-plane:6443/healthz?timeout=10s  in 0 milliseconds
2023-06-02T07:00:45.5662708Z [kubelet-check] It seems like the kubelet isn't running or healthy.
2023-06-02T07:00:45.5663494Z [kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get "http://localhost:10248/healthz": dial tcp [::1]:10248: connect: connection refused.
````

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -64,12 +64,18 @@ jobs:
         export SHELL=/bin/bash
         make build image
     - name: Start KinD
-      uses: engineerd/setup-kind@v0.5.0
+      uses: helm/kind-action@v1.7.0
```
Comment (ArthurSens) [inline .github/workflows/e2e.yaml:67]: Any special reason to change the action?

Comment (ArthurSens) [reply .github/workflows/e2e.yaml:67]: Well, just two comments above would explain it 🤦 

note to self, read the PR conversation before reading the changes :P

**Comment by simonpasquier** [comment]
Closing in favor of #5680

**Pull request closed without merging**